src/css/components/Button.scss

```diff
@@ -38,243 +36,95 @@ $Button-padding: 0 scaleGrid(2);
   height: $Button-height;
   line-height: $Button-line-height;
   margin: 0;
+  min-width: $Button-height;
   padding: $Button-padding;
   text-decoration: none;
   transition: all $duration-short ease-in-out;
   vertical-align: middle;
   white-space: nowrap;
 
   &:hover,
-  &:active,
   &:focus {
-    background-color: $Button-bg-hover;
-    border-color: $Button-border-color-hover;
-    color: $Button-color;
+    border-color: $Button-hover-color;
+    color: $Button-hover-color;
     outline: none;
   }
 
-  &.is-active {
-    background-color: $Button-bg-hover;
-    border-bottom-width: 1px;
-    border-top-width: $Button-border-bottom-width;
-    color: $Button-color;
+  &:active, &.is-active {
+    border-color: $Button-active-color;
+    color: $Button-active-color;
   }
 
   &.is-disabled,
   &:disabled {
     &,
     &:hover,
     &:active {
-      background-color: $Button-bg;
-      border-color: $Button-border-color;
+      background-color: color-neutral(2);
+      border-color: color-neutral(5);
       cursor: not-allowed;
-      color: $Button-color;
-      opacity: $Button-disabled-opacity;
+      color: color-neutral(5);
     }
   }
 }
 
+$permutations: (
+  "primary" color-primary(6) color-primary(5) color-primary(7),
+  "negative" color-negative(6) color-negative(5) color-negative(7),
+  "warning" color-warning(6) color-warning(5) color-warning(7),
+  "success" color-success(6) color-success(5) color-success(7)
+);
+
+@each $name, $color, $hover, $active in $permutations {
+  .Button--#{$name} {
+    background-color: $color;
+    border-color: $color;
+    color: $ts-white;
```

Comment:
I wonder if we want a colorNeutral(0) or something for white if we're not going to do the additional layer of abstraction with foreground, background, middleground, etc (which I support).

Reply:
Thoughts on color-neutral(1)? For TeamSnap's colors its #FBFBFB. Not sure if thats close enough, but my poor eyesight cant tell the difference.

Reply:
I can barely tell the difference visually, but it does drop it below the AAA standard for accessibility, which is 7.0. When the text is white it's 7.15, and at color-neutral(1) it's 6.91. I think ideally we would aim for the AAA standard when possible. I also don't know if we have any other plans for color-neutral(1) such that we could make it full white instead of fbfbfb. Any reason to not add the additional number 0 for white?

Looping in @andrewmeacham for his thoughts.

Reply:
Gonna go ahead and merge this in, but we can circle back to this at a later date if needed. This should be very easy to adapt if we decide we need a new color for it.